src/advection/SLAdvect.cpp

```diff
@@ -98,48 +98,48 @@ void SLAdvect::advect(Field *out, Field *in, const Field *u_vel, const Field *v_
             real Ck = dtz * d_w_vel[idx];
 
             // Calculation of horizontal indices and interpolation weights
-            long int i0;
-            long int i1;
-            real r;
+            long int i0 = i;
+            long int i1 = i;
+            real r = 1;
 
-            if (Ci > 0) {
+            if (Ci > epsilon_x) {
                 i0 = std::max(i_start, (i - static_cast<long int>(Ci)));
                 i1 = i0 - 1;
-                r = fabs(fmod(Ci, 1));
-            } else {
-                i1 = std::min(i_end, i - static_cast<long int>(Ci));
-                i0 = i1 + 1;
-                r = 1 - fabs(fmod(Ci, 1));
+                r = fmod(Ci, 1); // closer to i1 if r is closer to 1
+            } else if (Ci < -epsilon_x) {
+                i0 = std::min(i_end, i - static_cast<long int>(Ci));
+                i1 = i0 + 1;
+                r = fmod(-Ci, 1); // closer to i1 if r is closer to 1
```

Comment:
needs two spaces between code and comment

Reply:
not part of this PR anymore